worlds/kh2/__init__.py

```diff
@@ -78,232 +80,279 @@ def create_item(self, name: str, ) -> Item:
+
+        self.visitlocking_dict = Progression_Dicts["AllVisitLocking"].copy()
+        if self.multiworld.Schmovement[self.player] != "level_0":
+            for _ in range(self.multiworld.Schmovement[self.player].value):
+                for name in {ItemName.HighJump, ItemName.QuickRun, ItemName.DodgeRoll, ItemName.AerialDodge,
+                             ItemName.Glide}:
+                    self.item_quantity_dict[name] -= 1
+                    self.growth_list.remove(name)
+                    self.multiworld.push_precollected(self.create_item(name))
+
+        if self.multiworld.RandomGrowth[self.player] != 0:
+            max_growth = min(self.multiworld.RandomGrowth[self.player].value, len(self.growth_list))
+            for _ in range(max_growth):
+                random_growth = self.multiworld.per_slot_randoms[self.player].choice(self.growth_list)
+                self.item_quantity_dict[random_growth] -= 1
+                self.growth_list.remove(random_growth)
+                self.multiworld.push_precollected(self.create_item(random_growth))
+
+        if self.multiworld.Visitlocking[self.player] == "no_visit_locking":
+            for item, amount in Progression_Dicts["AllVisitLocking"].items():
+                for _ in range(amount):
+                    self.multiworld.push_precollected(self.create_item(item))
+                    self.item_quantity_dict[item] -= 1
+                    self.visitlocking_dict[item] -= 1
+                    if self.visitlocking_dict[item] == 0:
+                        self.visitlocking_dict.pop(item)
+
+        elif self.multiworld.Visitlocking[self.player] == "second_visit_locking":
+            for item in Progression_Dicts["2VisitLocking"]:
+                self.item_quantity_dict[item] -= 1
+                self.visitlocking_dict[item] -= 1
+                if self.visitlocking_dict[item] == 0:
+                    self.visitlocking_dict.pop(item)
+                self.multiworld.push_precollected(self.create_item(item))
+
+        for _ in range(self.multiworld.RandomVisitLockingItem[self.player].value):
+            if sum(self.visitlocking_dict.values()) <= 0:
+                break
+            visitlocking_set = list(self.visitlocking_dict.keys())
+            item = self.multiworld.per_slot_randoms[self.player].choice(visitlocking_set)
+            self.item_quantity_dict[item] -= 1
+            self.visitlocking_dict[item] -= 1
+            if self.visitlocking_dict[item] == 0:
+                self.visitlocking_dict.pop(item)
+            self.multiworld.push_precollected(self.create_item(item))
+
+        for item in item_dictionary_table:
+            data = self.item_quantity_dict[item]
+            for _ in range(data):
+                itempool.append(self.create_item(item))
+        # Creating filler for unfilled locations
+        while len(itempool) < self.totalLocations:
+            item = self.multiworld.per_slot_randoms[self.player].choice(self.filler_items)
```

Comment:
I would heavily recommend you overwrite `get_filler_item_name` and then call that or `self.create_filler()`. This would also have the added benefit of you having correct filler items whenever the multiworld generates filler items, such as item links. I do this in The Messenger if you want an example.